src/tuning.cpp

```diff
@@ -65,4 +67,97 @@ int Tuning::CalcPitchPos(int course, data_NOTATIONTYPE notationType, int lines)
     }
 }
 
+int Tuning::CalcPitchNumber(int course, int fret, data_NOTATIONTYPE notationType)
+{
+    // Use <tuning><course>s, if available
+    // else use @tuning.standard, if available
+    // else use @notationType
+    
+    // Do we have the tuning for this course?
+    AttNNumberLikeComparison cnc(COURSE, std::to_string(course));
+    Course* const courseTuning{vrv_cast<Course*>(FindDescendantByComparison(&cnc))};
+    
+    if (courseTuning && courseTuning->HasPname() && courseTuning->HasOct()) {
+
+        // Distance in semitones from the octave's starting C to the given note
+        int midiBase{0};
+        switch (courseTuning->GetPname()) {
+            case PITCHNAME_c: midiBase = 0; break;
+            case PITCHNAME_d: midiBase = 2; break;
+            case PITCHNAME_e: midiBase = 4; break;
+            case PITCHNAME_f: midiBase = 5; break;
+            case PITCHNAME_g: midiBase = 7; break;
+            case PITCHNAME_a: midiBase = 9; break;
+            case PITCHNAME_b: midiBase = 11; break;
+            case PITCHNAME_NONE: break;
+        }
+
+        const int octave{courseTuning->GetOct()};
+        
+        // As this does not represent historical notation of any kind the only
+        // accidentals we should ever see are "s" and "f"
+        const int alter{(courseTuning->HasAccid())
+            ? (courseTuning->GetAccid() == ACCIDENTAL_WRITTEN_s)
+                ? 1
+                : (courseTuning->GetAccid() == ACCIDENTAL_WRITTEN_f)
+                ? -1
+                : 0
+            : 0};
```

Comment:
I find this hard to read... Wouldn't it be clearer with some `if/else` or a `switch` statement?